flytekit/extras/tasks/shell.py

```diff
@@ -210,8 +217,22 @@ def execute(self, **kwargs) -> typing.Any:
             print(gen_script)
             print("\n==============================================\n")
 
+        if self._env is not None:
+            for k, v in self._env.items():
+                os.environ[k] = str(v)
+
+        working_dir = flytekit.current_context().working_directory
```

Comment:
https://github.com/flyteorg/flytekit/blob/386626356ff2cb810733a29a51cf3d6351c8424d/flytekit/core/utils.py#L170 has the source for this, looks like a pretty typically autodeleting temp dir, so look good to me. I think the only risk here is if the bash script being called for some reason looked for a tmp_script.sh (but unless someone passess in the temp directory path as an env variable, it seems unlikely they would know where to look for the tmp_script.sh anyway, so this seems extremely contrived)